src/main/java/com/esri/samples/editing/edit_and_sync_features/EditAndSyncFeaturesSample.java

```diff
@@ -0,0 +1,468 @@
+          // update button text and disable button
+          geodatabaseButton.setDisable(true);
+          geodatabaseButton.setText("Generating Geodatabase...");
+
+          // show progress indicator
+          progressIndicator.setVisible(true);
+
+          // disable updating the geodatabase extent
+          mapView.removeViewpointChangedListener(viewpointChangedListener);
+
+          // generate a geodatabase for the chosen area
+          generateGeodatabase();
+
+        } else if (currentEditState == EditState.READY) {
+          // sync the geodatabase
+          syncGeodatabase();
+        }
+      });
+
+      // handle clicks on the map view to select and move features
+      mapView.setOnMouseClicked((event) -> {
+
+        // return on mouse movement during click (i.e. click and drag)
+        if (!event.isStillSincePress()){
+          return;
+        }
+
+        if (event.getButton() == MouseButton.PRIMARY) {
+          // get screen point where user clicked
+          Point2D screenPoint = new Point2D(event.getX(), event.getY());
+
+          // get map location corresponding to screen point
+          Point mapPoint = mapView.screenToLocation(screenPoint);
+
+          // iterate through all feature layers in the map
+          mapView.getMap().getOperationalLayers().stream().filter(layer -> layer instanceof FeatureLayer).forEach(layer -> {
+            FeatureLayer featureLayer = (FeatureLayer) layer;
+
+            // start a process to identify any features at the clicked screen point
+            ListenableFuture<IdentifyLayerResult> results = mapView.identifyLayerAsync(featureLayer, screenPoint, 1, false, 1);
+            results.addDoneListener(() -> {
+              try {
+                // get the results of the identification process to determine whether a feature was clicked on
+                List<GeoElement> elementList = results.get().getElements();
+
+                // determine whether the clicked feature represents a point that can be moved
+                if (!elementList.isEmpty()){
+                  GeoElement geoElement = elementList.get(0);
+                  if (geoElement instanceof ArcGISFeature && geoElement.getGeometry().getGeometryType() == GeometryType.POINT) {
+                    // clear the previous selection (to avoid selecting multiple features at the same time)
+                    featureLayer.clearSelection();
+
+                    // grab the new feature and make it selected
+                    ArcGISFeature newSelectedFeature = (ArcGISFeature) elementList.get(0);
+                    featureLayer.selectFeature(newSelectedFeature);
+
+                    // change the state to editing
+                    currentEditState = EditState.EDITING;
+                  }
+
+                } else {
+                  // retrieve the selected features in the feature layer
+                  ListenableFuture<FeatureQueryResult> selectedQuery = featureLayer.getSelectedFeaturesAsync();
```

Comment:
Done. I didn't do this originally because I couldn't figure out how to de-select the feature without `featureLayer.clearSelection()`, i.e. knowing the feature layer it belongs to. Is there any other way to do this?

Reply:
No, you need to know the feature layer to deselect one of its features. In this case, `clearSelection` would be just as good because you're only selecting one feature at a time. If you wanted to deselect a specific future, use `unselectFeature(feature)`.

In editing workflows, you'd probably right-click on features to deselect them. In that case, you'd be able to get the feature layer from the identify result's layer content, and then use `unselectFeature`.